### ISSUE TITLE
pkg-config binary name is not overridable

### ISSUE THREAD

**Issue by rofl0r** [nodejs/node-v0.x-archive#25609]
in configure:

``` python
def pkg_config(pkg):
cmd = os.popen('pkg-config --libs %s' % pkg, 'r')
```

this breaks cross-compilation and user expectations, as it always uses the host pkg-config.
it is common practice to test the environment variable `PKG_CONFIG` and only if not set fall back to `pkg-config`.

a fix for this should be simple, ~2-5 lines of changed/added code

**Comment by jasnell** [comment]
@rofl0r ... interested in submitting a PR?

**Comment by rofl0r** [comment]
i'll consider it once my fix for #5197 is merged
